File: c-c/7.13.cpp
//7-13 日K蜡烛图 
/*
2021.10.12
16.51.30
By Xcl
*/
#include <stdio.h>

int main(){
    float a,b,c,d;
    scanf("%f %f %f %f",&a,&b,&c,&d);
    if(d<a){
        printf("BW-Solid");
        if(c<a&&c<d){
            printf(" with Lower Shadow");
            if(b>a&&b>d){
                printf(" and Upper Shadow");
            }
        }else{
            if(b>a&&b>d){
                printf(" with Upper Shadow");
            }
        }
    }else{
        if(d!=a){
            printf("R-Hollow");
            if(c<a&&c<d){
                printf(" with Lower Shadow");
                if(b>a&&b>d){
                    printf(" and Upper Shadow");
                }
            }else{
                if(b>a&&b>d){
                    printf(" with Upper Shadow");
                }
            }
        }else{
            printf("R-Cross");
            if(c<a&&c<d){
                printf(" with Lower Shadow");
                if(b>a&&b>d){
                    printf(" and Upper Shadow");
                }
            }else{
                if(b>a&&b>d){
                    printf(" with Upper Shadow");
                }
            }
        }
    }
}

File: python-p/7.13.py
##7-13 日K蜡烛图 
'''
2021.10.12
17.05.24
By Xcl
'''
a,b,c,d=map(float,input().split())
if(d<a):
    print('BW-Solid',end='')
    if(c<a and c<d):
        print(' with Lower Shadow',end='')
        if(b>a and b>d):
            print(' and Upper Shadow',end='')
    else:
        if(b>a and b>d):
            print(' with Upper Shadow',end='')
else:
    if(d!=a):
        print('R-Hollow',end='')
        if(c<a and c<d):
            print(' with Lower Shadow',end='')
            if(b>a and b>d):
                print(' and Upper Shadow',end='')
            else:
                if(b>a and b>d):
                    print(' with Upper Shadow',end='')
    else:
        print('R-Cross',end='')
        if(c<a and c<d):
            print(' with Lower Shadow',end='')
            if(b>a and b>d):
                print(' and Upper Shadow',end='')
        else:
            if(b>a and b>d):
                print(' with Upper Shadow',end='')
